massifs/localmassifreader.go

```diff
@@ -0,0 +1,384 @@
+// ReadMassifStart reads the massif start from the given log file
+func (r *LocalReader) ReadMassifStart(logfile string) (MassifStart, string, error) {
+	return r.cache.ReadMassifStart(logfile)
+}
+
+// EnsureReplicaDirs ensures the replica directories exist for the given tenant identity
+func (r *LocalReader) EnsureReplicaDirs(tenantIdentity string) error {
+	if !r.InReplicaMode() {
+		return fmt.Errorf("replica dir must be configured on the local reader")
+	}
+
+	massifsDir := filepath.Dir(r.GetMassifLocalPath(tenantIdentity, 0))
+	sealsDir := filepath.Dir(r.GetSealLocalPath(tenantIdentity, 0))
+
+	err := os.MkdirAll(massifsDir, os.FileMode(0755))
+	if err != nil {
+		return fmt.Errorf("%w: %s", ErrFailedToCreateReplicaDir, massifsDir)
+
+	}
+	err = os.MkdirAll(sealsDir, os.FileMode(0755))
+	if err != nil {
+		return fmt.Errorf("%w: %s", ErrFailedToCreateReplicaDir, sealsDir)
+
+	}
+	return nil
+}
+
+// GetVerifiedContext gets the massif and its seal and then verifies the massif
+// data against the seal. If the caller provides the expected public key, the
+// public key on the seal is required to match
+func (r *LocalReader) GetVerifiedContext(
+	ctx context.Context, tenantIdentity string, massifIndex uint64,
+	opts ...ReaderOption,
+) (*VerifiedContext, error) {
+
+	options, err := checkedVerifiedContextOptions(r.cache.Options().ReaderOptions, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	mc, err := r.GetMassif(ctx, tenantIdentity, massifIndex, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	return mc.verifyContext(ctx, options)
+}
+
+// GetHeadVerifiedContext gets the massif and its seal and then verifies the massif
+// data against the seal. If the caller provides the expected public key, the
+// public key on the seal is required to match
+func (r *LocalReader) GetHeadVerifiedContext(
+	ctx context.Context, tenantIdentity string,
+	opts ...ReaderOption,
+) (*VerifiedContext, error) {
+
+	options, err := checkedVerifiedContextOptions(r.cache.Options().ReaderOptions, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	mc, err := r.GetHeadMassif(ctx, tenantIdentity, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	return mc.verifyContext(ctx, options)
+}
+
+// VerifyContext verifies an arbitrary context and returns a verified context if this succeeds.
+func (r *LocalReader) VerifyContext(
+	ctx context.Context, mc MassifContext,
+	opts ...ReaderOption,
+) (*VerifiedContext, error) {
+
+	options, err := checkedVerifiedContextOptions(r.cache.Options().ReaderOptions, opts...)
+	if err != nil {
+		return nil, err
+	}
+	return mc.verifyContext(ctx, options)
+}
+
+// ReplaceVerifiedContext writes the content from the verified remote to the
+// local replica.  is the callers responsibility to ensure the context was
+// verified, and that the writeOpener opens the file in append mode if it
+// already exists.
+func (r *LocalReader) ReplaceVerifiedContext(
+	ctx context.Context, vc *VerifiedContext, writeOpener WriteAppendOpener,
```

Comment:
ctx isn't used in this function?

Reply:
fab, missed that. It might be I put it in so that an implementation of this interface could *have* a context. I'll try and convince myself that either this method isn't part of the interface or that it should never need a context